### PR TITLE
[SYCL][CUDA] Fix post merge errors from #9512

### PR DESCRIPTION
**Pull request by kbenzie**
Resolves the warnings as errors reported in [post merge](https://github.com/intel/llvm/actions/runs/5266121277/jobs/9519634360) as a result of merging #9512. Additionally move pre-processor guards to resolve unused global variables which would also fail in this build configuration (clang & SYCL_ENABLE_WERROR=ON).

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
@againull @smaslov-intel can you please review ASAP this post-commit failure fix?

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by dm-vodopyanov** [comment]
@steffenlarsen I guess if you have merge rights without code owners approvals, we can merge it.

**Pull request merged**
